ocm_python_wrapper/cluster.py

```diff
@@ -492,72 +525,61 @@ def validate_and_update_addon_parameters(self, user_parameters=None, use_api_def
-                conditions = param.get("conditions")
-                if param.get("required"):
-                    param_id = param["id"]
-                    default_value = param.get("default_value")
-                    if conditions:
-                        param_conditions = [
-                            condition["data"] for condition in conditions if condition["resource"] == "cluster"
-                        ]
-                        if param_conditions:
-                            for condition, condition_value in param_conditions[0].items():
-                                if not self.check_param_conditions(
-                                    clusters_dict=self.instance.to_dict(),
-                                    condition=condition,
-                                    condition_value=condition_value,
-                                ):
-                                    break
-                            else:
-                                _required_parameters[param_id] = {"default_value": default_value}
-                    else:
-                        _required_parameters[param_id] = {"default_value": default_value}
-            return _required_parameters
-
         _user_parameters = user_parameters or []
-        _info = self.addon_info()
-        addon_parameters = _info.get("parameters", {})
+        addon_parameters = self.addon_info().get("parameters", {})
         user_addon_parameters = [param["id"] for param in _user_parameters]
 
         if not addon_parameters and _user_parameters:
             raise ValueError(f"{self.addon_name} does not take any parameters, got {user_addon_parameters}")
 
-        required_parameters = _get_required_cluster_parameters(_addon_parameters=addon_parameters)
-        missing_parameter = []
+        addon_parameters_dict = self.get_addon_parameters_dict(_addon_parameters=addon_parameters)
+        _user_parameters = self.update_missing_params_from_defaults(
+            _user_parameters=_user_parameters,
+            addon_parameters_dict=addon_parameters_dict,
+            use_api_defaults=use_api_defaults,
+            user_addon_parameters=user_addon_parameters,
+        )
+
+        _user_parameters = self.update_param_value_type(
+            _user_parameters=_user_parameters,
+            addon_parameters_dict=addon_parameters_dict,
+        )
+
+        return _user_parameters
 
-        for param, param_dict in required_parameters.items():
-            if param not in user_addon_parameters:
-                if use_api_defaults and param_dict["default_value"]:
+    @staticmethod
+    def update_param_value_type(_user_parameters, addon_parameters_dict):
+        for param in _user_parameters:
+            param_type = addon_parameters_dict[param["id"]]["value_type"]
+            param_value = param["value"]
+            if not isinstance(param_value, param_type):
+                param["value"] = param_type(param_value)  # noqa: FCN001
```

Comment:
please remove `noqa: FCN001` and add to exclude list

Reply:
i do not think it will be clear there